### PR TITLE
feat: Moved /etxns/:id in View Expense to Platform

### PR DESCRIPTION
**Pull request by OmkarJ13**
### <samp>🤖 Generated by Copilot at 6cd151a</samp>

Refactored the view-expense page to use the new platform API for expenses. Renamed and updated the expense data model and observables to match the platform API response. Added a view parameter to toggle between spender and approver views. Removed unused types and enums.

### <samp>🤖 Generated by Copilot at 6cd151a</samp>

> _The platform API brings the doom_
> _We must align or face our tomb_
> _Renaming `etxn$` to `expense$`_
> _We refactor with no defense_

### <samp>🤖 Generated by Copilot at 6cd151a</samp>

*  Rename `etxn$` observable to `expense$` and `Expense` type to `PlatformExpense` to reflect new data model from platform API ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL23-R27), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL45-R46), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL57-R58), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL265-R269), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL274-R283), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL312-R316), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL365-R372), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL397-R402), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL416-R422), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL442-R443), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL484-R486), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL504-R505), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL609-R610), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL358-R360), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL374-R376), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL389-R391), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL404-R406), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL437-R437), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL446-R446), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL473-R473), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL537-R539), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL566-R566), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL586-R586), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL602-R602), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL641-R643), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL661-R661), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL679-R679), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL698-R698), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL715-R715), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL733-R733), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL800-R800), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL823-R823), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-717fc694a7c204cc8c4c0a0534c256936252ffa212d988754376936a1fe543a1L46-R48), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-717fc694a7c204cc8c4c0a0534c256936252ffa212d988754376936a1fe543a1L56-R58), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-717fc694a7c204cc8c4c0a0534c256936252ffa212d988754376936a1fe543a1L235-R247), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-717fc694a7c204cc8c4c0a0534c256936252ffa212d988754376936a1fe543a1L262-R288))
* Rename `etxn` variable to `expense` and update property names to match platform API response in `view-expense.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL23-R27), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL45-R46), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL57-R58), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL68-R75), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL81-R90), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL98-R101), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL105-R108), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL150-R152), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL164-R166), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL178-R180), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL200-R207), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL218-R220), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL256-R261), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL265-R269), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL274-R283), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL291-R295), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL312-R316), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL322-R326), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL337-R346), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL355-R359), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL365-R372), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL377-R386), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL391-R395), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL397-R402), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL416-R422), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL434-R435), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL442-R443), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL457-R463), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL475-R476), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL484-R486), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL491-R492), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL497-R498), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL504-R505), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL521-R527), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL539-R540), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL549-R552), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL563-R569), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL583-R589), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL601-R602), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL609-R610), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL618-R619), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL632-R633), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL646-R648))
* Remove `AccountType` enum from imports and use `source_account.type` property instead in `view-expense.page.ts` and `view-expense.page.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-717fc694a7c204cc8c4c0a0534c256936252ffa212d988754376936a1fe543a1L27), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL358-R360), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL374-R376), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL404-R406))
* Add `spenderExpensesService` and `approverExpensesService` to constructor parameters and use them to fetch expense data based on `view` parameter in `view-expense.page.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-717fc694a7c204cc8c4c0a0534c256936252ffa212d988754376936a1fe543a1L149-R153), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-717fc694a7c204cc8c4c0a0534c256936252ffa212d988754376936a1fe543a1L235-R247), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-717fc694a7c204cc8c4c0a0534c256936252ffa212d988754376936a1fe543a1R260-R261))
* Rename `etxnWithoutCustomProperties$` observable to `expenseWithoutCustomProperties$` and update property names to match platform API response in `view-expense.page.ts` and `view-expense.page.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL437-R437), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL446-R446), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL473-R473), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL586-R586), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL602-R602), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL641-R643), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL661-R661), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL679-R679), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-717fc694a7c204cc8c4c0a0534c256936252ffa212d988754376936a1fe543a1L56-R58), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-717fc694a7c204cc8c4c0a0534c256936252ffa212d988754376936a1fe543a1L262-R288))
* Update logic for checking `is_reimbursable` property to use false instead of true values in `view-expense.page.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL389-R391))
* Use `expenseState` pipe to transform state value to displayable format in `view-expense.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL105-R108))
* Use `category` object with nested properties instead of string in `view-expense.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL68-R75), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL256-R261), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL355-R359), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL397-R402), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL484-R486), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL491-R492), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL504-R505), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL549-R552))
* Use `report` object with nested properties instead of string in `view-expense.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL150-R152))
* Use `user` object with nested properties instead of string in `view-expense.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL164-R166))
* Use `tax_group` object with nested properties instead of string in `view-expense.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL274-R283))
* Use `cost_center` object with nested properties instead of string in `view-expense.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL322-R326), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL646-R648))
* Use `project` object with nested properties instead of string in `view-expense.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL337-R346), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL632-R633))
* Use `distance` and `distance_unit` properties instead of string in `view-expense.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL365-R372), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL377-R386), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL391-R395))
* Use `locations` array of objects with nested properties instead of array of strings in `view-expense.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL416-R422), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL434-R435), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL563-R569))
* Use `travel_classes` array of strings instead of string in `view-expense.page.html` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL497-R498), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL521-R527), [link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-38bc8efb9ce49f72845a76cfe9045d647245f157542e5281df732a05304ba5dcL539-R540))
* Use `matched_corporate_card_transactions` array of objects with nested properties instead of string in `view-expense.page.spec.ts` ([link](https://github.com/fylein/fyle-mobile-app/pull/2533/files?diff=unified&w=0#diff-3193ca2e160d90be32f2ebdc92d004542d0ad5042080accb5d69c8835a83f1feL566-R566))

## Clickup
https://app.clickup.com/t/8678z1xha

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/view-expense/view-expense.page.ts [hunk 1]
```diff
@@ -429,39 +432,6 @@ export class ViewExpensePage {
...
-
-      if (idx > -1) {
-        res = filename.substring(idx + 1, filename.length);
-      }
-    }
-
-    return res;
-  }
-
-  getReceiptDetails(file: FileObject): { type: string; thumbnail: string } {
```
Comment (OmkarJ13) [inline src/app/fyle/view-expense/view-expense.page.ts:447]: Not required anymore

File: src/app/fyle/view-expense/view-expense.page.ts [hunk 2]
```diff
@@ -392,24 +393,26 @@ export class ViewExpensePage {
...
-            fileObj.url = downloadUrl;
-            const details = this.getReceiptDetails(fileObj);
-            fileObj.type = details.type;
-            fileObj.thumbnail = details.thumbnail;
-            return fileObj;
+            const details = this.fileService.getReceiptsDetails(fileObj.name, downloadUrl);
+            const fileObjWithDetails: FileObject = {
+              url: downloadUrl,
+              type: details.type,
+              thumbnail: details.thumbnail,
```
Comment (OmkarJ13) [inline src/app/fyle/view-expense/view-expense.page.ts:412]: This is unused key, will be removing it from everywhere separately

**Review by SowmyaFyle — APPROVED** [review]

File: src/app/fyle/view-expense/view-expense.page.html [hunk 3]
```diff
@@ -42,8 +42,8 @@
 
 <ion-content>
   <ng-container *ngIf="!isLoading; else loading">
-    <div class="view-expense--container" *ngIf="etxn$|async as etxn">
-      <ng-container *ngIf="etxn.tx_policy_flag">
+    <div class="view-expense--container" *ngIf="expense$|async as expense">
```
Comment (hlkavya0213) [inline src/app/fyle/view-expense/view-expense.page.html:45]: use ngIf at the beginning

File: src/app/fyle/view-expense/view-expense.page.html [hunk 4]
```diff
@@ -65,27 +65,29 @@
...
+              {{ expense.amount | humanizeCurrency: expense.currency:true }}
             </span>
             <div *ngIf="isCCCTransaction">
               <div
                 class="view-expense--card-container__txn-type"
-                [ngClass]="{'view-expense--card-container__credit': etxn.tx_amount < 0}"
+                [ngClass]="{'view-expense--card-container__credit': expense.amount < 0}"
               >
-                {{ etxn.tx_amount > 0 ? 'DR' : 'CR' }}
+                {{ expense.amount > 0 ? 'DR' : 'CR' }}
```
Comment (hlkavya0213) [inline src/app/fyle/view-expense/view-expense.page.html:90]: don't we have the component to handle the debit/credit conventions ?

File: src/app/fyle/view-expense/view-expense.page.html [hunk 5]
```diff
@@ -95,15 +97,15 @@
           </div>
 
           <div class="view-expense--card-container__exchange-rate" *ngIf="foreignCurrencySymbol">
```
Comment (hlkavya0213) [inline src/app/fyle/view-expense/view-expense.page.html:99]: ngIf at the beginning pls, easier to read

**Review by hlkavya0213 — APPROVED** [review]
some comments

File: src/app/fyle/view-expense/view-expense.page.html (hunk 4 above)
Comment (OmkarJ13) [reply src/app/fyle/view-expense/view-expense.page.html:90]: Nopes, checked the mobile app codebase. We don't have a component for this, also this is only used in very few places, hence component might not be needed.

**Comment by OmkarJ13** [comment]
Unit tests are taken care off in separate PR, ignore the breaking tests here

File: src/app/fyle/view-expense/view-expense.page.html [hunk 6]
```diff
@@ -65,27 +65,29 @@
...
+          <div class="view-expense--card-container__category">
+            {{ expense.category?.display_name || 'Unspecified' }}
+          </div>
           <div class="view-expense--card-container__merchant">
-            {{ etxn.tx_vendor || merchantFieldName + ' Unspecified'}}
+            {{ expense.merchant || merchantFieldName + ' Unspecified'}}
           </div>
           <div *ngIf="isProjectShown" class="view-expense--card-container__project">
-            {{ projectFieldName | titlecase }}: {{ etxn.tx_project_name ? etxn.tx_project_name : 'Unspecified' }}
+            {{ projectFieldName | titlecase }}: {{ expense.project?.name || 'Unspecified' }}
```
Comment (abhishek1234321) [inline src/app/fyle/view-expense/view-expense.page.html:75]: shouldn't we just show `-`? What do we show in webapp?

**Review by abhishek1234321 — APPROVED** [review]

File: src/app/fyle/view-expense/view-expense.page.html (hunk 6 above)
Comment (OmkarJ13) [reply src/app/fyle/view-expense/view-expense.page.html:75]: In web app, we show `-`. But mobile app shows `Unspecified` in view expense pages

**Pull request merged**